x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/conditional/SpatialIntersects.java

```diff
@@ -0,0 +1,274 @@
+                                ? SpatialIntersectsGeoPointDocValuesAndStringEvaluator.Factory::new
+                                : SpatialIntersectsGeoPointDocValuesAndSourceEvaluator.Factory::new
+                        )
+                    );
+                    evaluatorMap.put(
+                        SpatialEvaluatorKey.fromSourceAndConstant(spatialType, otherType),
+                        new SpatialEvaluatorWithConstantFactory(SpatialIntersectsGeoPointDocValuesAndConstantEvaluator.Factory::new)
+                    );
+                }
+            }
+        }
+
+        // Support cartesian_point and cartesian_shape from source and constant combinations
+        for (DataType spatialType : new DataType[] { CARTESIAN_POINT, CARTESIAN_SHAPE }) {
+            for (DataType otherType : new DataType[] { CARTESIAN_POINT, CARTESIAN_SHAPE, DataTypes.KEYWORD }) {
+                evaluatorMap.put(
+                    SpatialEvaluatorKey.fromSources(spatialType, otherType),
+                    new SpatialEvaluatorFactoryWithFields(
+                        // Both Geometry and String fields are backed by BytesRef, so we need different evaluators
+                        otherType == DataTypes.KEYWORD
+                            ? SpatialIntersectsCartesianSourceAndStringEvaluator.Factory::new
+                            : SpatialIntersectsCartesianSourceAndSourceEvaluator.Factory::new
+                    )
+                );
+                evaluatorMap.put(
+                    SpatialEvaluatorKey.fromSourceAndConstant(spatialType, otherType),
+                    new SpatialEvaluatorWithConstantFactory(SpatialIntersectsCartesianSourceAndConstantEvaluator.Factory::new)
+                );
+                if (EsqlDataTypes.isSpatialPoint(spatialType)) {
+                    evaluatorMap.put(
+                        SpatialEvaluatorKey.fromSources(spatialType, otherType).withDocValues(),
+                        new SpatialEvaluatorFactoryWithFields(
+                            // Both Geometry and String fields are backed by BytesRef, so we need different evaluators
+                            otherType == DataTypes.KEYWORD
+                                ? SpatialIntersectsCartesianPointDocValuesAndStringEvaluator.Factory::new
+                                : SpatialIntersectsCartesianPointDocValuesAndSourceEvaluator.Factory::new
+                        )
+                    );
+                    evaluatorMap.put(
+                        SpatialEvaluatorKey.fromSourceAndConstant(spatialType, otherType),
+                        new SpatialEvaluatorWithConstantFactory(SpatialIntersectsCartesianPointDocValuesAndConstantEvaluator.Factory::new)
+                    );
+                }
+            }
+        }
+
+        // If both sides are string fields or constants we will treat generically
+        SpatialEvaluatorKey key = SpatialEvaluatorKey.fromSources(DataTypes.KEYWORD, DataTypes.KEYWORD);
```

Comment:
I'm wondering if we should require the right hand side to be parsed rather than evaluate strings directly.

Reply:
With, like, `to_geo_point` or whatever.

Reply:
I did a review of PostGIS, and they support literal strings. My goal has been to conform to PostGIS functionality whenever possible or reasonable, with the aim of benefitting from PostGIS deep penetration into the market. I plan to do a presentation on the similarities and differences between Spatial-ESQL and PostGIS for us to discuss in a team meeting.

Reply:
We could infer the conversion and add it rather than push it into the evaluator.

Reply:
If we inferred a call to `TO_GEOSHAPE` for example, that would accomplish the first step of the conversion, but we still need to do the DocValues and Component2D conversions. Still this is worth learning more, where would such inference be undertaken?

Reply:
There's kind of an "easy way" and a "better way". I'd suggest the easy way for now - basically just build the evaluator for the conversion when you build the other evaluator. That's not able to share the conversions, but it does give you a starting point.